pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -301,3 +301,26 @@ func (a *acrProvider) parseACRLoginServerFromImage(image string) string {
 
 	return ""
 }
+
+// shouldUseCache decide whether to use cache according to image and cache content
+func shouldUseCache(image string, d credentialprovider.DockerConfig) bool {
+	if len(d) > 2 {
+		// for service principal(len=5), always use cache
+		return true
+	}
+
+	match := acrRE.FindAllString(image, -1)
+	if len(match) != 1 {
```

Comment:
Do you need to handle the similar custom cloud case (eg, AzureStack) as parseACRLoginServerFromImage?

Reply:
@northtyphoon I think yes, do you know whether ACR login server would have `.azurecr.` in AzureStack, custom cloud case needs cloud provider specific environment, don't want to change the common provider interface, otherwise it may get rejected.

Reply:
/cc: @xizhamsft

Reply:
I think checking `.azurecr.` should be OK. We have added such logic in #90425.

Reply:
it's done, PTAL, thanks. Also updated the unit test cases.